### ISSUE TITLE
Documentation: Negative Binomial

### ISSUE THREAD

**Issue by fullerf** [JuliaStats/Distributions.jl#474]
Reading the documentation on the Univariate Discrete > NegativeBinomial, I was led to think that the wikipedia definition of the distribution was being used.  Upon testing it appears that the "Mathematica definition" is used instead.  You'll get good agreement between the observed mean and the formula: E[NB(n,p)] = n(1-p)/p. Also tested Var[NB(n,p)] = n(1-p)/p^2.  Both of these agreed well with the implementation in Distributions.jl.

The difference between the Wikipedia definition and the "Mathematica Definition", is just a change of variables: NB_wiki(n,1-p) = NB_mathematica(n,p).  For the wiki definition, the mean has the formula np/(1-p), variance = np/(1-p)^2.

tl;dr: Documentation doesn't quite match implementation.  It's off by a change of variables.

**Comment by johnmyleswhite** [comment]
A pull request that improves the documentation would very helpful.

**Comment by sivapvarma** [comment]
@fullerf The definition in [documenation](http://distributionsjl.readthedocs.io/en/latest/univariate-discrete.html#NegativeBinomial) says the samething as mathematica's documentation [here](https://reference.wolfram.com/language/ref/NegativeBinomialDistribution.html). 

Both Julia and Mathematica's implementation  count the number of failures before _r_ th success with _p_ as probability of success in an individual trial. 

I guess the confusion was due to subtle difference in pmf's between the two definitions.

For Julia/Mathematica definition `pmf(k)` will be proportional to `p^r (1-p)^k` whereas for the Wikipedia definition(number of success until _r_ failures) `pmf(k)` will be proportional to `p^k (1-p)^r`

**Comment by mschauer** [comment]
This problem could be alleviated by means of keyword arguments via #823, with named arguments such as `psuccess` or `pfailure` in the constructor.

**Comment by matbesancon** [comment]
I would consider this closed by #825
